tla-io/src/test/scala/at/forsyte/apalache/io/itf/TestItfToTla.scala

```diff
@@ -0,0 +1,480 @@
+        )
+    )
+    val xyRecT = RecT1("x" -> IntT1, "y" -> StrT1)
+
+    assert {
+      itfToTla.typeDrivenBuild(xyRec, IntT1).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(xyRec, RecT1()).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(xyRec, RecT1("x" -> IntT1, "y" -> StrT1, "z" -> IntT1)).isLeft
+    }
+
+    assert(itfToTla
+          .typeDrivenBuild(xyRec, xyRecT)
+          .map(_.build)
+          .contains(tla
+                .rec(
+                    "x" -> tla.int(1),
+                    "y" -> tla.str("abc"),
+                )
+                .build))
+
+    // case TupT1(elems @ _*) =>
+
+    val tupOneA = UJsonRep(
+        Obj(
+            ItfToTla.TUP_FIELD ->
+              Arr(1, "A")
+        )
+    )
+
+    val tupT = TupT1(IntT1, StrT1)
+
+    assert {
+      itfToTla.typeDrivenBuild(one, tupT).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(tupOneA, SetT1(IntT1)).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(tupOneA, TupT1()).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(tupOneA, TupT1(IntT1, StrT1, BoolT1)).isLeft
+    }
+
+    assert(itfToTla.typeDrivenBuild(tupOneA, tupT).map(_.build).contains(tla.tuple(tla.int(1), tla.str("A")).build))
+
+    // case SetT1(elemT) =>
+
+    val emptySet = UJsonRep(Obj(ItfToTla.SET_FIELD -> Arr()))
+
+    val setT = SetT1(BoolT1)
+
+    assert {
+      itfToTla.typeDrivenBuild(one, setT).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(emptySet, IntT1).isLeft
+    }
+
+    assert(itfToTla.typeDrivenBuild(emptySet, setT).map(_.build).contains(tla.emptySet(setT.elem).build))
+
+    val boolSet = UJsonRep(Obj(ItfToTla.SET_FIELD -> Arr(true, false)))
+
+    assert {
+      itfToTla.typeDrivenBuild(boolSet, SetT1(IntT1)).isLeft
+    }
+
+    assert(itfToTla
+          .typeDrivenBuild(boolSet, setT)
+          .map(_.build)
+          .contains(tla.enumSet(tla.bool(true), tla.bool(false)).build))
+
+    // case FunT1(argT, resT) =>
+
+    val emptyFun = UJsonRep(Obj(ItfToTla.MAP_FIELD -> Arr()))
+
+    val funT = FunT1(IntT1, IntT1)
+
+    assert {
+      itfToTla.typeDrivenBuild(one, funT).isLeft
+    }
+
+    assert {
+      itfToTla.typeDrivenBuild(emptyFun, IntT1).isLeft
+    }
+
+    assert(itfToTla
+          .typeDrivenBuild(emptyFun, funT)
+          .map(_.build)
+          .contains(tla
+                .setAsFun(tla.emptySet(TupT1(funT.arg, funT.res)))
+                .build))
+
+    val id12 = UJsonRep(
+        Obj(ItfToTla.MAP_FIELD ->
+          Arr(Arr(1, 1), Arr(2, 2)))
+    )
+
+    assert {
+      itfToTla.typeDrivenBuild(id12, FunT1(IntT1, StrT1)).isLeft
+    }
+
+    assert(itfToTla
+          .typeDrivenBuild(id12, funT)
+          .map(_.build)
+          .contains(tla
+                .setAsFun(tla.enumSet(
+                        Seq(1, 2).map { i => tla.tuple(tla.int(i), tla.int(i)) }: _*
+                    ))
+                .build))
+
+  }
+
+  test("getTrace") {
```

Comment:
I don't see any tests here to check that the correct behavior occurs in the presence of junk fields, like `{"#set": [], "foo": "bar"}`.

Can you add a named test case for this (and similar edge cases, if you think of any?).